toolkit/tools/pkgworker/pkgworker.go

```diff
@@ -83,17 +79,15 @@ func main() {
 	defines[rpm.DistroReleaseVersionDefine] = *distroReleaseVersion
 	defines[rpm.DistroBuildNumberDefine] = *distroBuildNumber
 
-	err = retry.Run(func() error {
-		err = buildSRPMInChroot(chrootDir, rpmsDirAbsPath, *workerTar, *srpmFile, *repoFile, *rpmmacrosFile, defines, *noCleanup, *runCheck)
-		if err != nil {
-			logger.Log.Warnf("Failed package build attempt (%v), error (%v)", *srpmFile, err)
-		}
-		return err
-	}, *retryAttempts, retryDuration)
-	logger.PanicOnError(err, "Failed to build SRPM '%s'. For details see log file: %s.", *srpmFile, *logFile)
+	builtRPMs, err := buildSRPMInChroot(chrootDir, rpmsDirAbsPath, *workerTar, *srpmFile, *repoFile, *rpmmacrosFile, defines, *noCleanup, *runCheck)
+	logger.PanicOnError(err, "Failed to build SRPM '%s'. For details see log file: %s .", *srpmFile, *logFile)
 
 	err = copySRPMToOutput(*srpmFile, srpmsDirAbsPath)
 	logger.PanicOnError(err, "Failed to copy SRPM '%s' to output directory '%s'.", *srpmFile, rpmsDirAbsPath)
+
+	// On success write a comma-seperated list of RPMs built to stdout that can be parsed by the invoker.
+	// Any output from logger will be on stderr so stdout will only contain this output.
+	fmt.Printf(strings.Join(builtRPMs, delimiter))
```

Comment:
>strings.Join(builtRPMs, delimiter) [](start = 12, length = 34)

Nit: here's I'd argue it might be easier to read this if we provide the delimiter directly. It should be easy to understand what `string.Join` does even, if you're using it for the first time. Alternatively to stick to one convention at least within this function, we could create a const for `strings.TrimSuffix` above.